kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -251,6 +261,39 @@ public void publish(Engine engine) throws PublishFailedException {
     ((CatalogCommitter) committer).publish(engine, publishMetadata);
   }
 
+  @Override
+  public void writeChecksum(Engine engine, Snapshot.ChecksumWriteMode mode) throws IOException {
+    final Snapshot.ChecksumWriteMode actual = getStatistics().getChecksumWriteMode();
+
+    if (actual != mode) {
+      logger.warn("Requested {} mode but actual mode is {}.", mode, actual);
+    }
+
+    switch (mode) {
+      case NONE:
+        logger.info("Skipping writing checksum file: input mode was NONE");
+        return;
+      case SIMPLE:
+        final CRCInfo crcInfo =
+            logReplay
+                .getCrcInfoAtSnapshotVersion()
```

Comment:
Just curious, do you know why we store this here? Had to do some digging to understand how/when we populate this